code/modules/logging/log_holder.dm

```diff
@@ -61,10 +78,120 @@ GLOBAL_PROTECT(logger)
+			continue
+
+		if(!initial(category_type.schema_version))
+			stack_trace("log category type '[category_type]' does not have a valid schema version, skipping")
+			continue
+
+		var/master_category = initial(category_type.master_category)
+		if(master_category)
+			sub_categories[master_category] += list(category_type)
+			continue
+		category_tree[category_type] = list()
+
+	// Sort the sub categories
+	for(var/datum/log_category/master as anything in sub_categories)
+		if(!(master in category_tree))
+			stack_trace("log category [master] is an invalid master category as its a sub category")
+			continue
+		for(var/datum/log_category/sub_category as anything in sub_categories[master])
+			if(initial(sub_category.secret) != initial(master.secret))
+				stack_trace("log category [sub_category] has a secret status that differs from its master category [master]")
+			category_tree[master] += list(sub_category)
+
+	return category_tree
+
+/// Initializes the given log category and populates the list of contained categories based on the sub category list
+/datum/log_holder/proc/init_log_category(datum/log_category/category_type, list/datum/log_category/sub_categories)
+	var/datum/log_category/category_instance = new category_type
+
+	var/list/contained_categories = list()
+	for(var/datum/log_category/sub_category as anything in sub_categories)
+		sub_category = new sub_category
+		var/sub_category_actual = sub_category.category
+		sub_category.master_category = category_instance
+		log_categories[sub_category_actual] = sub_category
+
+		if(!semver_to_list(sub_category.schema_version))
+			stack_trace("log category [sub_category_actual] has an invalid schema version '[sub_category.schema_version]'")
+			sub_category.schema_version = LOG_CATEGORY_SCHEMA_VERSION_NOT_SET
+
+		contained_categories += sub_category_actual
+
+	log_categories[category_instance.category] = category_instance
+
+	if(!semver_to_list(category_instance.schema_version))
+		stack_trace("log category [category_instance.category] has an invalid schema version '[category_instance.schema_version]'")
+		category_instance.schema_version = LOG_CATEGORY_SCHEMA_VERSION_NOT_SET
+
+	contained_categories += category_instance.category
+
+	var/list/category_header = list(
+		LOG_HEADER_INIT_TIMESTAMP = logging_start_timestamp,
+		LOG_HEADER_ROUND_ID = big_number_to_text(GLOB.round_id),
+		LOG_HEADER_SECRET = category_instance.secret,
+		LOG_HEADER_CATEGORY_LIST = contained_categories,
+		LOG_HEADER_CATEGORY = category_instance.category,
+	)
+
+	rustg_file_write("[json_encode(category_header)]\n", category_instance.get_output_file(null))
+	if(human_readable_enabled)
+		rustg_file_write("\[[human_readable_timestamp()]\] Starting up round ID [round_id].\n - -------------------------\n", category_instance.get_output_file(null, "log"))
+
+/datum/log_holder/proc/unix_timestamp_string() // pending change to rust-g
+	return RUSTG_CALL(RUST_G, "unix_timestamp")()
```

Comment:
Are those changes to rust-g still pending?

Reply:
annoyingly yes